Holovibes/sources/API.cc

```diff
@@ -1624,6 +1616,28 @@ void set_record_mode(const std::string& text)
     }
 }
 
+void set_record_mode(const std::string& text)
+{
+    LOG_FUNC(text);
+
+    // Mapping from string to RecordMode
+    static const std::unordered_map<std::string, RecordMode> recordModeMap = {{"Chart", RecordMode::CHART},
+                                                                              {"Processed Image", RecordMode::HOLOGRAM},
+                                                                              {"Raw Image", RecordMode::RAW},
+                                                                              {"3D Cuts XZ", RecordMode::CUTS_XZ},
+                                                                              {"3D Cuts YZ", RecordMode::CUTS_YZ},
+                                                                              {"Moments", RecordMode::MOMENTS}};
+
+    auto it = recordModeMap.find(text);
+    if (it == recordModeMap.end())
+    {
+        LOG_ERROR("Unknown record mode {}", text);
+        throw std::runtime_error("Record mode not handled");
+    }
+
+    set_record_mode_enum(it->second);
+}
+
```

Comment:
If you reorder the fields you can delete this function and rename the `_enum` one

Reply:
I originally renamed the set_record_mode taking the RecordMode enum as an argument because in the .hxx, the setter for the setting RecordMode has the same signature

Reply:
Okay keep it like that, in the future i will merge the two functions

Reply:
alright
I still deleted the string function tho